### ISSUE TITLE
Substraction of a Datetime object from an Array{Datetime}

### ISSUE THREAD

**Issue by scheidan** [quinnj/Datetime.jl#22]
There seems to a problem with the substraction of a `Datetime` object from an `Array{Datetime}`:

``` julia
t1 = datetime(2000, 11, 1)
t_array = [datetime(2001, 11, 1) : month(7) : datetime(2005, 11, 1)]

## --- works as expected
t_array[1] - t1
t1 .- t_array
t1 - t_array

## --- however:
t_array - t1
## ERROR: i not defined
##  in - at C:\Users\scheidan\.julia\Datetime\src\Datetime.jl:358
## WARNING: backtraces on your platform are often misleading or partially incorrect

t_array .- t1
## ERROR: i not defined
##  in .- at operators.jl:83
## WARNING: backtraces on your platform are often misleading or partially incorrect
```

I'm using Julia 0.2.0-RC4, 64bit on Windows 7.

Hope that helps.

**Comment by quinnj** [comment]
Sorry for the delay on this. Yeah, there must be some kind of bug. I have a few reworks in progress, so I'll try to work it into those branches to push.

**Comment by quinnj** [comment]
Ok, I figured this out and it works on my latest master rewrite that's going into Base. See JuliaLang/julia#5328 for the code.

**Comment by quinnj** [comment]
Should be fixed by ed8123ede2e316a181a781ecd54fd735657b6cd8
